### PR TITLE
Fixed latex export for svg data in python 3

### PR DESCRIPTION
**Pull request by MSeal**
Fixes https://github.com/jupyter/nbconvert/issues/981

Verified that the rendered latex output matches python 2 output visually. This will probably fix other python exports outside of latex svg given the code path involved in the fix.

### CODE REVIEW

File: nbconvert/preprocessors/extractoutput.py [hunk 1]
```diff
@@ -80,27 +80,28 @@ def preprocess_cell(self, cell, resources, cell_index):
                 if mime_type in out.data:
                     data = out.data[mime_type]
 
-                    if (
-                        not isinstance(data, text_type)
-                        or mime_type == 'application/json'
-                    ):
+                    if mime_type == 'application/json':
```
Comment (akhmerov) [inline nbconvert/preprocessors/extractoutput.py]: This is too strict and will break some data types; compare with even a partial listing here: https://jupyter.readthedocs.io/en/latest/reference/mimetype.html

Also this change invalidates the follow-up comment, please keep them in sync.

**Comment by akhmerov** [comment]
In the original implementation I've relied on the nbformat spec that guaranteed that json data is parsed directly and not serialized.

I don't actually understand why that bit requires changing to support svg: IIUC svg should be of text-type and shouldn't be treated as json by that code.

File: nbconvert/preprocessors/extractoutput.py (hunk 1 above)
Comment (MSeal) [reply nbconvert/preprocessors/extractoutput.py]: Fair point on the strictness.

**Comment by MSeal** [comment]
If you only add the svg test I added you'll see if fails for python 3 in the same way the original issue describes. The `data` object coming back from `out.data[mime_type]` is definitely `bytes` and adding `data = data.decode('utf-8')`, or even the original `data = json.dumps(data)`, generically to all byte data before any other processing seemed wrong.

I'll look into the test case I added and see if there's a less strict way to solve the issue.

**Comment by akhmerov** [comment]
At the very least lists/dicts should be detected as definitely json

**Comment by MSeal** [comment]
I looked into the failing pattern, it's when the mimetype is `application/pdf` that it fails in the test case originally.

I changed the implementation to mirror the original behavior more closely, with just logically moving the base64 data type checks in front of the json / text checks. Can you take a look again @akhmerov for any issue you can see?

**Comment by MSeal** [comment]
Thanks for the review!

**Pull request merged**